pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,189 @@
 { lib
 , stdenv
+, fetchFromGitHub
+, fetchYarnDeps
 , fetchurl
+, rustPlatform
 , autoPatchelfHook
-, dpkg
+, wrapGAppsHook
+, esbuild
 , glib-networking
 , gst_all_1
 , libappindicator
 , libayatana-appindicator
+, nodejs
+, openssl
+, pkg-config
+, yq-go
+, pnpm-lock-export
+, prefetch-yarn-deps
 , webkitgtk
-, wrapGAppsHook
+, yarn
 }:
 
-stdenv.mkDerivation (finalAttrs: {
-  name = "dorion";
-  version = "4.1.3";
+let
+
+  pname = "dorion";
+  version = "4.1.0";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-O6KXOouutrNla5dkHRQeT0kp8DQO9MLoJrIMuqam/60=";
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-s1EQ2DcvxkPh0w3PwfNqzYDlW1vuWXsZ+WgYKO8mDMM=";
+
+    nativeBuildInputs = [ pnpm-lock-export ];
+    postFetch = ''
+      (cd $out && pnpm-lock-export --schema yarn.lock@v1)
```

```suggestion
      pushd $out
      pnpm-lock-export --schema yarn.lock@v1
      popd
```
I am not sure if the subshell could avoid error handling

---

Also with #290715 we now have porper pnpm support, maybe that helps